### ISSUE TITLE
debugWindow on window with unicode title crashes

### ISSUE THREAD

**Issue by dminuoso** [xmonad/xmonad-contrib#422]
### Problem Description

With Firefox 83.0 (64-bit), go visit an arbitrary web page, and run debugWindow on the window (say via debugStackString or debugStackFullString).

The resulting String has a bottom hidden inside that trips exactly on the  `— ` character.
```
Dec 11 01:57:44 Verleihnix xmonad[13387]: Enum.toEnum{Word8}: tag (8212) is outside of bounds (0,255)
```

I'm guessing the error is the call to `decodeString`.
https://github.com/xmonad/xmonad-contrib/blob/master/XMonad/Util/DebugWindow.hs#L124
If the buffer is already UTF8 encoded, this would blow up on code points outside the ASCII range.

The specialization of toEnum{Word8} would also fit (I don't see other code paths with going toEnum into Word8).

Because I'm struggling with other things, I couldn't pin point this with `+RTS xc -RTS` yet.

**Comment by dminuoso** [comment]
```when (enc == uTF8_STRING) $ error "String is not UTF8_STRING"```
```when (enc == cOMPOUND_TEXT) $ error "String is not COMPOUND_TEXT"```
This looks... wrong...

**Comment by liskin** [comment]
fixed by #423
